src/languageservice/services/yamlCompletion.ts

```diff
@@ -144,11 +144,11 @@ export class YamlCompletion {
 
     this.arrayPrefixIndentation = '';
     let overwriteRange: Range = null;
+    const isOnlyHyphen = lineContent.match(/^\s*(-)\s*$/);
```

Comment:
I am not sure if doing this with RegEx is a good idea. For instance this fails on a line like `  -   # a comment `. Can we use the AST to check if the rest of the line does not have any semantically significant tokens?

Reply:
it was there from the previous implementation, I just reused it (moved it up). I didn't want to change it.

But agree, 
I went a little bit deeper and there is another problem.

The real problem is in `getNodeFromPosition`
This method doesn't give proper results for arrays and maybe some others. That is the reason why there has to be custom code to 'fix' proper node based on special conditions 'empty lines, arrays, ...' (lines 343 to 515 in yamlCompletion)

for example, this yaml gives these results:
foo:
  - # foo object is returned (should be foo[0])
    # foo object is returned (should be foo[0])
    item1: aaaf
    # foo[0] object is returned (OK)
  - # foo object is returned (should be foo[1])
    # foo[!!0!!] object is returned (should be foo[1])
    item2: bbb
    # foo[1] object is returned (OK)

So this PR doesn't make sense anymore (it's possible to reuse tests)

The question is whether your team has the capacity to fix it.
If not I will try it in the future, but don't have much time for it :-/